tests/live_mode/reorg.rs

```diff
@@ -0,0 +1,116 @@
+};
+
+use crate::{
+    common::{TestCounter, build_provider, deploy_counter, spawn_anvil},
+    mock_callbacks::BlockOrderingCallback,
+};
+use alloy::{
+    eips::BlockNumberOrTag, network::Ethereum, providers::ext::AnvilApi, sol_types::SolEvent,
+};
+use event_scanner::{event_filter::EventFilter, event_scanner::EventScannerBuilder};
+
+#[tokio::test]
+async fn reorg_rescans_events_with_rewind_depth() -> anyhow::Result<()> {
+    let anvil = spawn_anvil(0.1).unwrap();
+    let provider = build_provider(&anvil).await.unwrap();
+
+    let contract = deploy_counter(provider.clone()).await?;
+    let contract_address = *contract.address();
+    let blocks = Arc::new(Mutex::new(Vec::new()));
+    let callback = Arc::new(BlockOrderingCallback { blocks: Arc::clone(&blocks) });
+    let filter = EventFilter {
+        contract_address: Some(contract_address),
+        event: Some(TestCounter::CountIncreased::SIGNATURE.to_owned()),
+        callback,
+    };
+
+    let mut scanner = EventScannerBuilder::new()
+        .with_event_filter(filter)
+        .with_reorg_rewind_depth(6)
+        .connect_ws::<Ethereum>(anvil.ws_endpoint_url())
+        .await?;
+
+    tokio::spawn(async move { scanner.start(BlockNumberOrTag::Latest, None).await });
+
+    let mut expected_event_block_number = vec![];
+
+    let initial_events = 5;
+    for _ in 0..initial_events {
+        let receipt = contract.increase().send().await.unwrap().get_receipt().await.unwrap();
+        expected_event_block_number.push(receipt.block_number.unwrap());
+    }
+
+    let blocks_clone = Arc::clone(&blocks);
+    let event_counting = async move {
+        while blocks_clone.lock().await.len() < initial_events {
+            sleep(Duration::from_millis(100)).await;
+        }
+    };
+
+    if timeout(Duration::from_secs(1), event_counting).await.is_err() {
+        bail!("expected {initial_events} events, got {}", blocks.lock().await.len());
+    }
+
+    let mut tx_block_pairs = vec![];
+    let num_new_events = 3;
+    let latest_block = provider.get_block_by_number(BlockNumberOrTag::Latest).await?.unwrap();
+    let reorg_depth = 10;
+    for _ in 0..num_new_events {
+        let tx = contract.increase().into_transaction_request();
+        // 0 is offset to reorg depth
+        tx_block_pairs.push((TransactionData::JSON(tx), 0));
+        // pushing to latest block - roerg depth + 1;
+        expected_event_block_number.push(latest_block.header.number - reorg_depth + 1);
+    }
+    let reorg_options = ReorgOptions { depth: reorg_depth, tx_block_pairs };
+
+    provider.anvil_reorg(reorg_options).await.unwrap();
+    let new_latest_block =
+        provider.get_block_by_number(BlockNumberOrTag::Latest).full().await?.unwrap();
+    // sanity checks, block numb stays the same but hash changes
+    assert_eq!(new_latest_block.header.number, latest_block.header.number);
+    assert_ne!(new_latest_block.header.hash, latest_block.header.hash);
+    let new_block = provider
```

Comment:
```suggestion

    let new_block = provider
```
nit: add new line, increase clarity